components/public-api/gitpod/v1/workspace.proto

```diff
@@ -38,6 +132,233 @@ message WatchWorkspaceStatusResponse {
   WorkspaceStatus status = 2;
 }
 
+message ListWorkspacesRequest {
+  enum OrderByField {
+    ORDER_BY_FIELD_UNSPECIFIED = 0;
+    // ORDER_BY_FIELD_CREATION_TIME is the default sorting field, it sorts
+    // workspaces by (instances) creation time
+    ORDER_BY_FIELD_CREATION_TIME = 1;
+  }
+
+  // pagination contains the pagination options for listing workspaces
+  PaginationRequest pagination = 1;
+
+  // order_by is the field to sort workspaces by, allowed values are OrderBy
+  OrderBy order_by = 2;
+
+  // scope of the workspaces
+  WorkspaceScope scope = 3;
+
+  // organization_id is the ID of the organization that contains the workspaces
+  string organization_id = 4;
+
+  // pinned indicates whether to list only pinned workspaces
+  bool pinned = 5;
+
+  // search_term is a search term to filter workspaces by name
+  string search_term = 6;
+
+  string workspace_id = 7;
+
+  string owner_id = 8;
+}
+
+message ListWorkspacesResponse {
+  // workspaces are the workspaces that matched the query
+  repeated Workspace workspaces = 1;
+
+  // pagination contains the pagination options for listing workspaces
+  PaginationResponse pagination = 2;
+}
+
+message CreateAndStartWorkspaceRequest {
+  message Source {
```

Comment:
here we wanted to have another design, let's hold migration create to the end, i.e. see https://www.notion.so/gitpod/Stable-Public-API-Project-Spec-f5225e862cc94b22a659094d1e7a4570?pvs=4#c17532fbbf884e0798c66abb9d6d93b6